File: 0038.报数/0038-报数.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 1) return "1";
        if (n == 2) return "11";
        int i = 2, j = 0;
        string temp = "";
        string res = "11";
        
        for (i; i <n; ++ i ){ // No. i
            j = 0;
            temp = "";
            for (j; j < res.size(); ){
                int k = 0;
                while(j + k < res.size() && res[j] == res[j+k])
                    k += 1;
                temp += to_string(k) + res[j];
                // cout << res[j]<<endl;
                // cout << to_string(res[j] - '0')<<endl;
                // cout <<temp<<endl;
                j+= k;
            }
            res = temp;
            // cout <<res<<endl;
        } 
        return res;
        
    }
};

File: 0038.报数/0038-报数.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        record = ["1"]
        for i in range(1, n):
            pre = record[i - 1]
            idx = 0
            tmp = ""
            while idx < len(pre):
                cnt = 1
                while(idx + 1 < len(pre) and pre[idx] == pre[idx + 1]):
                    idx += 1
                    cnt += 1
                tmp += str(cnt) + pre[idx]
                idx += 1
            record.append(tmp)
        return record[-1]